File: registers/inp_register/INPRegister.py
from m5.params import *
from m5.SimObject import SimObject

class INPRegister(SimObject):
    type = 'INPRegister'
    cxx_header = "CAProject/registers/inp_register/inp_register.hh"
    cxx_class = "gem5::INPRegister"

    inp = Param.UInt8(0, "Input stored in the register")

File: registers/inp_register/inp_register.cc
#include "CAProject/registers/inp_register/inp_register.hh"

namespace gem5
{

INPRegister::INPRegister(const INPRegisterParams &params) : 
    SimObject(params),
    inp(params.inp)
{
}

void
INPRegister::setValue(uint8_t val)
{
    inp = val;
}

uint8_t
INPRegister::getValue()
{
    return inp;
}

} // namespace gem5
